app/api/login/route.ts

```diff
@@ -0,0 +1,21 @@
+import { OAuth2Client } from "google-auth-library";
+import { NextRequest, NextResponse } from "next/server";
+const client = new OAuth2Client();
+
+export async function POST(request: NextRequest) {
+    const body = await request.json();
+    const clientId = process.env.GOOGLE_LOGIN_CLIENT_ID;
+    const ticket = await client.verifyIdToken({
+        idToken: body.token,
+        audience: clientId
+    });
+    const payload = ticket.getPayload();
+    const userObj = {
+        email: payload?.email,
+        name: {
+            firstName: payload?.given_name,
+            lastName: payload?.family_name
+        }
+    }
```

Comment:
fft: I guess we probably do always have these props, however I am wondering what this looks like in the long term. How does the site stay logged in? And how/when are we re decoding these tokens? Do we always pass back this info? Is this info we should persist? Not sure any of these need to be answered right now, but we should be careful to make sure that our backend and front end stay in sync and we don't end up with some weird mismatched expectations.

Reply:
_resolving as the conversation was moved [here](https://github.com/jjm3x3/AmazingRaceFantasy/pull/285#discussion_r2072504820)_